File: apps/server/internal/models/provider.go
package models

import (
	"encoding/json"
	"time"

	"github.com/google/uuid"
)

// Provider is a catalog entry for a storage provider type (Google Drive,
// OneDrive, Dropbox, S3-compatible, ...). It is global, not tenant-scoped:
// individual connections live in storage_accounts.
type Provider struct {
	ID           uuid.UUID       `db:"id" json:"id"`
	Slug         string          `db:"slug" json:"slug"`
	Name         string          `db:"name" json:"name"`
	Protocol     string          `db:"protocol" json:"protocol"`
	AuthType     string          `db:"auth_type" json:"auth_type"`
	Capabilities json.RawMessage `db:"capabilities" json:"capabilities,omitempty"`
	IsActive     bool            `db:"is_active" json:"is_active"`
	DeletedAt    *time.Time      `db:"deleted_at" json:"deleted_at,omitempty"`
	CreatedAt    time.Time       `db:"created_at" json:"created_at"`
	UpdatedAt    time.Time       `db:"updated_at" json:"updated_at"`
}

File: apps/app/lib/api/models/provider.ts
import type { ISO8601DateString } from '@/types/time'

export interface Provider {
  id: string
  slug: string
  name: string
  protocol: string
  auth_type: string
  capabilities?: Record<string, unknown>
  is_active: boolean
  deleted_at?: ISO8601DateString | null
  created_at: ISO8601DateString
  updated_at: ISO8601DateString
}
